src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,127 @@
+package duke.parser;
+
+import duke.exceptions.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.storage.Storage;
+import duke.ui.Ui;
+
+
+public class Parser {
+    private Storage storage;
+    private Ui ui;
+
+    public Parser(Storage storage, Ui ui) {
+        this.storage = storage;
+        this.ui = ui;
+    }
+
+    public boolean parse(String input) {
+        String cmd;
+
+        try{
+            String inputs[] = input.split(" ", 2);
+            cmd = inputs[0];
+
+            int idx;
+            Task tsk;
+            String markMsg, tskNum;
+            String cfmMsg = "Got it. I've added this duke.exceptions.task:";
+            String[] desc;
+            LocalDateTime date, from, to;
+
+            TaskList data = this.storage.loadTasks();
+
+            switch (cmd) {
```

Comment:
Maybe the input handlers can be abstracted as different classes with separate methods for easier future extension?